jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPAProcessInstanceManager.java

```diff
@@ -160,7 +160,11 @@ public ProcessInstance getProcessInstance(long id, boolean readOnly) {
                 kruntime.getProcessInstance(parentProcessInstanceId);
             }
             processInstance.setKnowledgeRuntime( kruntime );
+            
             ((ProcessInstanceImpl) processInstance).reconnect();
+            if (readOnly) {
+                internalRemoveProcessInstance(processInstance);
```

Comment:
Just a short question - why is the process instance removed if it is read only and if I see correctly it is not added to the processInstance map anywhere in the code?

Reply:
@MarianMacik because it should not be there, it should always be loaded from the store as it should not receive any updates as they won't be persisted.